pagefaultgames/pokerogue
# [Ability] Implement ability illusion

## What are the changes?

Adding illusion

## Why am I doing these changes?



Make ability illusion works so we can play Zoroark and make fusion with illusion

## What did change?


Added PreSummonAbAttr class as ability attribute.
A pokemon with illusion create an illusion from the last pokemon of the trainer's party.
If we encounter a wild Zoroark he will use the illusion of Entei, Suicune or Raiku like in the original game. We can change that in the method generateIllusion() in src/field/pokemon.ts.
The class pokemon has a new attribute illusion. 
A shiny Zoroark will create a shiny illusion.
A non shiny Zoroark will create a non-shiny illusion.
This doesn't affect the luck of course, it's just visual.
A pokemon with illusion can use illusion one time per battle like in the original game.
After a battle, the pokemon can use illusion again.
The AI move choice will be affected by the illusion.

### Screenshots/Videos


https://discord.com/channels/1125469663833370665/1176874654015684739/1251234015760089159
https://discord.com/channels/1125469663833370665/1176874654015684739/1251234045204103238
https://discord.com/channels/1125469663833370665/1176874654015684739/1251234047104127039

## How to test the changes?




I played Zoroark and override the enemy pokemon with Zoroark so we can see him using illusion of Raikou Suicune Entei
I used illusion on a fusion and with a fusion
I tried to switch with the illusion

## Checklist
